### PR TITLE
feat: add a step command to verify if a URL returns an expected HTTP status code

### PR DESCRIPTION
**Pull request by ccojocar**
This command waits with exponential backoff for a URL to return an expected status code. It will
fail when the timeout is exceeded.

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

Add a new command `step verify url` which waits with exponential backoff for a URL to return an expected HTTP status code. 

This is particularly useful to check when a URL configured by the external DNS is ready during the boot sequence.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/cmd/step/verify/step_verify_url.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return nil
+}
+
+func (o *StepVerifyURLOptions) logError(err error) error {
+	log.Logger().Infof("Retrying due to: %s", err)
+	return err
+}
+
+// Run waits with exponential backoff for an endpoint to return an expected HTTP status code
+func (o *StepVerifyURLOptions) Run() error {
```
Comment (cagiti) [inline pkg/cmd/step/verify/step_verify_url.go]: Should we a utility function here or re-use an existing backoff strategy?

Comment (ccojocar) [reply pkg/cmd/step/verify/step_verify_url.go]: It's using a utility function. See L91.

**Pull request merged**